spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientBootstrapConfiguration.java

```diff
@@ -37,29 +35,24 @@
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnKubernetesEnabled
 @AutoConfigureAfter(KubernetesBootstrapConfiguration.class)
+@Import({ KubernetesCommonsAutoConfiguration.class, KubernetesClientAutoConfiguration.class })
 public class KubernetesClientBootstrapConfiguration {
 
-	@Configuration(proxyBeanMethods = false)
-	@Import({ KubernetesCommonsAutoConfiguration.class, KubernetesClientAutoConfiguration.class })
-	protected static class KubernetesPropertySourceConfiguration {
```

Comment:
- remove the inner class
- replace `@ConditionalOnProperty(name = "spring.cloud.kubernetes.config.enabled", matchIfMissing = true)` with `@ConditionalOnKubernetesConfigEnabled`. 

This is actually a backport change that we already have done in fabric8 case.